accounts/abi/bind/bindv2_test.go

```diff
@@ -0,0 +1,390 @@
+		nil,
+		nil,
+		v1TestBindingNameConflict,
+	},
+	{
+		"RangeKeyword",
+		[]string{`[{"inputs":[{"internalType":"uint256","name":"range","type":"uint256"}],"name":"functionWithKeywordParameter","outputs":[],"stateMutability":"pure","type":"function"}]`},
+		[]string{"0x608060405234801561001057600080fd5b5060dc8061001f6000396000f3fe6080604052348015600f57600080fd5b506004361060285760003560e01c8063527a119f14602d575b600080fd5b60436004803603810190603f9190605b565b6045565b005b50565b6000813590506055816092565b92915050565b600060208284031215606e57606d608d565b5b6000607a848285016048565b91505092915050565b6000819050919050565b600080fd5b6099816083565b811460a357600080fd5b5056fea2646970667358221220d4f4525e2615516394055d369fb17df41c359e5e962734f27fd683ea81fd9db164736f6c63430008070033"},
+		nil,
+		nil,
+		v1TestBindingRangeKeyword,
+	},
+	{
+		"NumericMethodName",
+		[]string{`[{"anonymous":false,"inputs":[{"indexed":false,"internalType":"address","name":"_param","type":"address"}],"name":"_1TestEvent","type":"event"},{"inputs":[],"name":"_1test","outputs":[],"stateMutability":"pure","type":"function"},{"inputs":[],"name":"__1test","outputs":[],"stateMutability":"pure","type":"function"},{"inputs":[],"name":"__2test","outputs":[],"stateMutability":"pure","type":"function"}]`},
+		[]string{"0x6080604052348015600f57600080fd5b5060958061001e6000396000f3fe6080604052348015600f57600080fd5b5060043610603c5760003560e01c80639d993132146041578063d02767c7146049578063ffa02795146051575b600080fd5b60476059565b005b604f605b565b005b6057605d565b005b565b565b56fea26469706673582212200382ca602dff96a7e2ba54657985e2b4ac423a56abe4a1f0667bc635c4d4371f64736f6c63430008110033"},
+		nil,
+		nil,
+		v1TestBindingNumericMethodName,
+	},
+}
+
+func TestBindingV2ConvertedV1Tests(t *testing.T) {
+	os.Mkdir("convertedv1bindtests", 0777)
+	for _, tc := range bindTests2 {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.types == nil {
+				tc.types = []string{tc.name}
+			}
+
+			code, err := bind(&tc)
+			if err != nil {
+				t.Fatalf("got error from bind: %v", err)
+			}
+
+			// TODO: remove these before merging abigen2 PR.  these are for convenience if I need to regenerate the converted bindings or add a new one.
+			if err := os.WriteFile(fmt.Sprintf("convertedv1bindtests/%s.go", strings.ToLower(tc.name)), []byte(code), 0666); err != nil {
```

Comment:
This shouldn't be here, should it -- writing the file out? 
@fjl mentioned a similar case, he uses an ENV var to determine "are we in test-writing mode or consume-mode?". Another option is to make one test TestWriteBindings, which has `t.Skip("this test generates testdata")` and needs to be explicitly enabled to run.

Reply:
Yeah, this shouldn't be writing anything as part of the test case.  I just included the code to write out the generated bindings for convenience (because they would change a lot as I made modifications on this PR).

Somehow the code to write out the generated bindings got uncommented and committed. 

I've pushed changes so that the test case is now written as I intend it for merging.

Reply:
> @fjl mentioned a similar case, he uses an ENV var to determine "are we in test-writing mode or consume-mode?". Another option is to make one test TestWriteBindings, which has t.Skip("this test generates testdata") and needs to be explicitly enabled to run.

Interesting.  I guess I figured that test-writing logic like this wouldn't be included in the codebase at all.  But I guess it kind of makes sense, and we could copy the logic from the test-case you mentioned to apply it here.